src/realmd/AuthSocket.cpp

```diff
@@ -511,6 +513,13 @@ bool AuthSocket::_HandleLogonProof()
     if (!Read((char*)&lp, sizeof(sAuthLogonProof_C)))
         return false;
 
+    PINData pinData;
+    if(lp.securityFlags)
+    {
+        if(!Read((char*)&pinData, sizeof(pinData)))
+            return false;
+    }
+
```

Comment:
no a fan of the idea of turning the two if statements into a single one? or not possible within this scope?